frontend/src/bundles/chat/components/message-list/message-list.tsx

```diff
@@ -0,0 +1,65 @@
+import { Flex, VStack } from '~/bundles/common/components/components.js';
+import { UserAvatar } from '~/bundles/users/components/components.js';
+
+import { MessageSender } from '../../enums/enums.js';
+import { type Message, type MessageGroup } from '../../types/types.js';
+import { MessageBox } from '../message-box/message-box.js';
+
+type Properties = {
+    messages: Message[];
+};
+
+const MessageList: React.FC<Properties> = ({ messages }) => {
+    const groupedMessages: MessageGroup[] = [];
+    let currentGroup: MessageGroup | null = null;
+
+    for (const message of messages) {
+        const { sender } = message;
+        if (!currentGroup || currentGroup.sender !== sender) {
+            currentGroup = { sender: sender, messages: [message] };
+            groupedMessages.push(currentGroup);
+        } else {
+            currentGroup.messages.push(message);
+        }
+    }
```

Comment:
I think it would be better to wrap this logic in useMemo.
Also, have you tried using .reduce() instead? I guess It will be more compact

Reply:
Yes try to used reduce, but when making the commit the lint said it cant be used 

25:38  error  `Array#reduce()` is not allowed  unicorn/no-array-reduce, the useMemo will be applied